### ISSUE TITLE
semantic highlight and hints for a TypedDict's get function.

### ISSUE THREAD

**Issue by amotzop** [microsoft/pylance-release#4098]
## Environment data

-   Language Server version: v2023.3.20
-   OS and version: ubuntu 20.04
-   Python version (& distribution if applicable, e.g. Anaconda): 3.9.5

## Code Snippet
```python
from typing_extensions import TypedDict

class Cls(TypedDict):
    a: int
    b: str

dct: Cls = {"a": 1, "b": "2"}
dct.get("a")
```

## Repro Steps
1. Look at the semantic highlighting of the above code.
2. Put your curser on the `get` part to see the type hint.

## Expected behavior
1. The `get` function should be colored as such.
2. The hint should show the various possible signatures like for a regular overloaded function (where it shows a single signature and you can move between the different signatures manually).

## Actual behavior
1. The `get` function is uncolored.
2. The hint looks like `get: Overload[(k: Literal['a']) -> int, (k: Literal['a'], default: Any) -> int, (k: Literal['b']) -> str, (k: Literal['b'], default: Any) -> str, (k: str) -> (Any | None), (k: str, default: __TDefault@Cls) -> (Any | __TDefault@Cls)])`.

**Comment by erictraut** [comment]
The problem is that the `get` symbol for a `TypedDict` class is completely synthesized by the type checker. It has no declaration. This is handled internally by specifying a `_synthesizedType` in the associated symbol, but that symbol has no associated declaration. The hover provider and semantic token provider currently operate on declarations. They both have fallback logic to handle symbols without declarations in a reasonable manner, but this fallback logic is different from the mainstream logic. 

I've updated the hover provider in the pyright code base to better handle this case.

The semantic token provider in the pylance code base will likewise need to be updated to handle the case where there are no declarations but there is a synthesized type associated with the symbol.

**Comment by kikones34** [comment]
Any updates on this? It appears it's still not behaving as expected.

**Comment by rchiodo** [comment]
I believe it's still being discussed on what to do:
https://github.com/microsoft/pyright/issues/9252

**Comment by KacieKK** [comment]
This issue has been fixed in prerelease version 2024.11.100, which we've just released. You can find the changelog here: [CHANGELOG.md](https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#202411100-6-November-2024-PreRelease)
